lib/streamlit/delta_generator.py

```diff
@@ -355,16 +356,41 @@ def wrapper(*args: Any, **kwargs: Any) -> NoReturn:
         return wrapper
 
     @property
-    def _parent_block_types(self) -> Iterable[str]:
+    def _parent_block_types(self) -> Iterable[Tuple[str, Optional[int]]]:
         """Iterate all the block types used by this DeltaGenerator and all
         its ancestor DeltaGenerators.
         """
         current_dg: Optional[DeltaGenerator] = self
         while current_dg is not None:
             if current_dg._block_type is not None:
-                yield f"{current_dg._block_type}:{current_dg._root_container}"
+                yield current_dg._block_type, current_dg._root_container
             current_dg = current_dg._parent
 
+    def _get_raw_parent_block_types(
+        self, parent_block_types: Iterable[Tuple[str, Optional[int]]]
+    ) -> FrozenSet[str]:
+        parent_block_types_raw = []
+        for parent_block, _ in parent_block_types:
+            parent_block_types_raw.append(parent_block)
+        return frozenset(parent_block_types_raw)
+
+    def _count_num_of_parent_columns(
+        self, parent_block_types: Iterable[Tuple[str, Optional[int]]]
+    ) -> int:
+        num_of_columns = 0
+        for parent_block, _ in parent_block_types:
+            if "column" == parent_block:
+                num_of_columns = num_of_columns + 1
+        return num_of_columns
+
+    def _check_if_one_of_parents_is_a_sidebar(
```

Comment:
nit: I like it when predicted (functions that return bool) start with `is`, so I'd propose renaming to: `_is_parent_a_sidebar`